keps/sig-node/5018-dra-adminaccess/README.md

```diff
@@ -0,0 +1,687 @@
+   spec:
+     resourceClassName: admin-resource-class
+     adminAccess: true
+   ```
+
+1. Namespace Label for DRA Admin Mode:
+
+   Namespaces must be labeled to allow the creation of admin-access claims.
+
+   Example label:
+
+   ```yaml
+   metadata:
+     labels:
+       kubernetes.io/dra-admin-access: "true"
+   ```
+
+1. Authorization Check:
+
+   In the REST storage layer, validate requests to create `ResourceClaim` or
+   `ResourceClaimTemplate` objects with `adminAccess: true`. Only authorize if
+   namespace has the `kubernetes.io/dra-admin-access` label.
+
+1. Grants privileged access to the requested device:
+
+   For requests with `adminAccess: true`, the scheduler bypasses standard
+   allocation checks and allows administrators to access devices already in use.
+   This ensures privileged tasks like monitoring or diagnostics can be performed
+   without disrupting existing allocations.
+
+1. No impact on availability of claims:
+
+   The scheduler ignores claims with `adminAccess: true`, normal usage is not
+   impacted as claims in other namespaces can still be allocated using the same
+   devices that are also accessed by workloads in the admin namespace.
+
+### Workflow
+
+1. A cluster administrator labels an admin namespace with
+   `kubernetes.io/dra-admin-access`.
+
+1. Only authorized users can create `ResourceClaim` or `ResourceClaimTemplate`
+   objects with `adminAccess: true` in this admin namespace.
+
+1. Only users with access to the admin namespace can reference these
+   `ResourceClaims` or `ResourceClaimTemplates` in their pod or deployment
+   specs.
+
+1. The DRA controller processes the request and grants privileged access to the
+   requested device.
+
+## Design Details
+
+### API Changes
+
+The `adminAccess` field in `DeviceRequest` was added to `ResourceClaim` and
+`ResourceClaimTemplate` in Kubernetes v1.31:
+
+```go
+// DeviceRequest is a request for devices required for a claim.
+// This is typically a request for a single resource like a device, but can
+// also ask for several identical devices.
+//
+// A DeviceClassName is currently required. Clients must check that it is
+// indeed set. It's absence indicates that something changed in a way that
+// is not supported by the client yet, in which case it must refuse to
+// handle the request.
+type DeviceRequest struct {
+	...
+
+    // AdminAccess indicates that this is a claim for administrative access
+    // to the device(s). Claims with AdminAccess are expected to be used for
+    // monitoring or other management services for a device.  They ignore
+    // all ordinary claims to the device with respect to access modes and
+    // any resource allocations.
+    //
+    // This is an alpha field and requires enabling the DRAAdminAccess
+    // feature gate. Admin access is disabled if this field is unset or
+    // set to false, otherwise it is enabled.
+    //
+    // +optional
+    // +featureGate=DRAAdminAccess
+    AdminAccess *bool
```

Comment:
Reading this doc, we seem to refer to "AdminAccess" as privileged for most of this doc. Have we considered just using Privileged instead of AdminAccess?

Reply:
I think it came up. If there's consensus that "Privileged" has clear advantages, then we have an opportunity to rename in 1.33 because we are doing a v1beta2 where we can use the new field name. We also would need to rename the KEP and feature gate or accept some inconsistency.

Reply:
IMO AdminAccess is better because it signals this field is meant for admins to use and manage, not just any privileged user on the cluster and should be used with caution.

Reply:
Looking at other examples in our pod API we do seem to use privileged: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L7704-#L7709

Reply:
AFAIK There’s nothing in-tree that prevents securityContext privileged from being set by non-admins. It can be gated by using PSA labels or VAP or policy webhooks. IMO “adminAccess” signals that this is an admin field and can only be used in an admin namespace. I don’t think the privileged field has the same effect. 

and if we were to rename the field now, do we need to support both old and new at the same time before we remove the old one in a future release? seems like extra complexity we need to maintain. 

Do we have consensus to rename? thoughts? @enj @liggitt @deads2k @pohly

Reply:
I prefer to not rename.